File: code/sample.py
"""
sample.py — Synthetic Python module for parser and syntax testing.
Demonstrates common patterns: dataclasses, enums, type hints, comprehensions.
"""

from __future__ import annotations

import statistics
from dataclasses import dataclass, field
from enum import Enum, auto
from typing import Optional


# ── Enums ─────────────────────────────────────────────────────────────────────

class Department(Enum):
    ENGINEERING = auto()
    MARKETING = auto()
    FINANCE = auto()
    HR = auto()


# ── Data model ────────────────────────────────────────────────────────────────

@dataclass(frozen=True)
class Employee:
    id: int
    name: str
    department: Department
    salary: float
    active: bool = True


# ── Repository ────────────────────────────────────────────────────────────────

@dataclass
class EmployeeRepository:
    _employees: list[Employee] = field(default_factory=list)

    def add(self, employee: Employee) -> None:
        self._employees.append(employee)

    def find_by_id(self, employee_id: int) -> Optional[Employee]:
        return next((e for e in self._employees if e.id == employee_id), None)

    def find_active(self) -> list[Employee]:
        return [e for e in self._employees if e.active]

    def group_by_department(self) -> dict[Department, list[Employee]]:
        result: dict[Department, list[Employee]] = {}
        for emp in self._employees:
            result.setdefault(emp.department, []).append(emp)
        return result

    def average_salary(self) -> float:
        salaries = [e.salary for e in self._employees]
        return statistics.mean(salaries) if salaries else 0.0

    def top_earners(self, n: int = 3) -> list[Employee]:
        return sorted(self._employees, key=lambda e: e.salary, reverse=True)[:n]


# ── Entry point ───────────────────────────────────────────────────────────────

def main() -> None:
    repo = EmployeeRepository()
    sample_data = [
        Employee(1, "Alice Martin",  Department.ENGINEERING, 82000),
        Employee(2, "Bob Chen",      Department.MARKETING,   64000),
        Employee(3, "Carol White",   Department.ENGINEERING, 91000),
        Employee(4, "David Kim",     Department.HR,          57000, active=False),
        Employee(5, "Eva Rossi",     Department.ENGINEERING, 78000),
        Employee(6, "Frank Lopez",   Department.MARKETING,   61500),
        Employee(7, "Grace Müller",  Department.FINANCE,     73000, active=False),
        Employee(8, "Henry Tanaka",  Department.HR,          55000),
    ]
    for emp in sample_data:
        repo.add(emp)

    print(f"Active employees : {len(repo.find_active())}")
    print(f"Average salary   : ${repo.average_salary():,.2f}")
    print("Top 3 earners    :", [e.name for e in repo.top_earners(3)])

    by_dept = repo.group_by_department()
    for dept, members in by_dept.items():
        print(f"  {dept.name:<15} {len(members)} member(s)")


if __name__ == "__main__":
    main()

File: code/sample.java
import java.util.*;
import java.util.stream.*;

/**
 * Sample Java class for parser and syntax testing.
 * Demonstrates common Java patterns: generics, streams, records, enums.
 */
public class sample {

    // ── Enum ──────────────────────────────────────────────────────────────────

    public enum Department {
        ENGINEERING, MARKETING, FINANCE, HR;
    }

    // ── Record (Java 16+) ─────────────────────────────────────────────────────

    public record Employee(int id, String name, Department department, double salary, boolean active) {}

    // ── Repository ────────────────────────────────────────────────────────────

    public static class EmployeeRepository {

        private final List<Employee> employees;

        public EmployeeRepository(List<Employee> employees) {
            this.employees = Collections.unmodifiableList(employees);
        }

        public List<Employee> findActive() {
            return employees.stream()
                    .filter(Employee::active)
                    .collect(Collectors.toList());
        }

        public Map<Department, List<Employee>> groupByDepartment() {
            return employees.stream()
                    .collect(Collectors.groupingBy(Employee::department));
        }

        public OptionalDouble averageSalary() {
            return employees.stream()
                    .mapToDouble(Employee::salary)
                    .average();
        }

        public Optional<Employee> findById(int id) {
            return employees.stream()
                    .filter(e -> e.id() == id)
                    .findFirst();
        }
    }

    // ── Main ──────────────────────────────────────────────────────────────────

    public static void main(String[] args) {
        List<Employee> data = List.of(
            new Employee(1, "Alice Martin",  Department.ENGINEERING, 82000, true),
            new Employee(2, "Bob Chen",      Department.MARKETING,   64000, true),
            new Employee(3, "Carol White",   Department.ENGINEERING, 91000, true),
            new Employee(4, "David Kim",     Department.HR,          57000, false),
            new Employee(5, "Eva Rossi",     Department.ENGINEERING, 78000, true),
            new Employee(6, "Frank Lopez",   Department.MARKETING,   61500, true),
            new Employee(7, "Grace Müller",  Department.FINANCE,     73000, false),
            new Employee(8, "Henry Tanaka",  Department.HR,          55000, true)
        );

        EmployeeRepository repo = new EmployeeRepository(data);

        System.out.println("Active employees: " + repo.findActive().size());
        repo.averageSalary().ifPresent(avg ->
            System.out.printf("Average salary: %.2f%n", avg));
        repo.findById(3).ifPresent(e ->
            System.out.println("Found: " + e.name() + " (" + e.department() + ")"));
    }
}
